v1.6.0 release note. [skip ci] (#7746)
dmlc/xgboost: NEWS.md

developments. The `approx` should see significant performance gain for many datasets as
mentioned in previous section, while the `hist` tree method also enjoys improved
performance with the removal of the internal `pruner` along with some other
refactoring. Lastly, `gpu_hist` no longer synchronize the device during training. (#7737)

### General bug fixes
* Fixes in CMake script for exporting configuration. (#7730)
* XGBoost can now handle unsorted sparse input. This includes text file format like libsvm
  and scipy sparse matrix where column index is not sorted. (#7731)
* Fix tree param feature type, this affects inputs with number of columns greater than the
maximum value of int32. (#7565)
* Fix external memory with gpu_hist and subsampling. (#7481)
* Check number of trees in inplace predict, this avoids a potential segfault when an
  incorrect value for `iteration_range` is provided. (#7409)

### Changes in the Python package
Other than the changes in Dask, the XGBoost Python package gained some new features and
improvements along with small bug fixes.

* Python 3.7 is required as the lowest Python version. (#7682)

> * Binary package Support Apple Silicon. (#7621, #7612)

Reviewer: ```suggestion
* Pre-built binary wheel for Apple Silicon. (#7621, #7612) Apple Silicon users will now be able to run `pip install xgboost` to install XGBoost.
```

Can you also merge in https://github.com/dmlc/xgboost/pull/7747 as part of 1.6 release?

format is set to UBJSON, which affects Python pickle and R RDS. (#7572, #7570, #7358,
#7571, #7556, #7549, #7416)

### General new features and improvements
Aside from the major new features mentioned above, some others are summarized here:

* Users can now access the build information of XGBoost binary in Python and C
  interface. (#7399, #7553)
* Auto-configuration of `seed_per_iteration` is removed, now distributed training should
  generate closer results to single node training when sampling is used. (#7009)
* A new parameter `huber_slope` is introduced for the `Pseudo-Huber` objective.
* During source build, XGBoost can choose cub in the system path automatically. (#7579)
* XGBoost now honors the CPU counts from CFS, which is usually set in docker
  environments. (#7654, #7704)
* The metric `aucpr` is rewritten for better performance and GPU support. (#7297, #7368)
* Metric calculation is now performed in double precision. (#7364)
* XGBoost no longer mutates the global OpenMP thread limit. (#7537, #7519, #7608, #7590,
  #7589, #7588, #7687)
* The default behavior of `max_leave` and `max_depth` is now unified (#7302, #7551).
* CUDA fat binary is now compressed. (#7601)

> * Deterministic result for evaluation metric and linear model In previous versions of

Author: ```suggestion
* Deterministic result for evaluation metric and linear model. In previous versions of
```

(#7365, #7736, #7607, #7574, #7521, #7514, #7456, #7453, #7455, #7434, #7429, #7405, #7381)

### External memory support
External memory support for both approx and hist tree method is considered feature
complete in XGBoost 1.6.  Building upon the iterator-based interface introduced in the
previous version, now both `hist` and `approx` iterates over each batch of data during
training and prediction.  In previous versions, `hist` concatenates all the batches into
an internal representation, which is removed in this version.  As a result, users can
expect higher scalability in terms of data size but might experience lower performance due
to disk IO. (#7531, #7320, #7638, #7372)

### Rewritten approx

The `approx` tree method is rewritten based on the existing `hist` tree method, the
rewrite closes the feature gap between `approx` and `hist` and improves the performance.
Now the behavior and `approx` should be more aligned with `hist` and `gpu_hist`, here's a
list of user-visible changes:

- Supports both `max_leaves` and `max_depth`.
- Supports `grow_policy`.
- Supports monotonic constraint.
- Supports feature weights.
- Use `max_bin` to replace `sketch_eps`.

> - Supports for categorical data.

Author: ```suggestion
- Support for categorical data.
```

This file records the changes in xgboost library in reverse chronological order.

## v1.6.0 (2022 Apr 16)

After a long period of development, XGBoost v1.6.0 is packed with many new features and
improvements. We summarize them in the following sections starting with an introduction to
some major new features, then moving on to language binding specific changes including new
features and notable bug fixes for that binding.

### Development of categorical data support
This version of XGBoost features new improvements and full coverage of experimental
categorical data support in Python and C package with tree model.  Both `hist`, `approx`

> and `gpu_hist` now supports training with categorical data.  Also, partition-based

Reviewer: ```suggestion
and `gpu_hist` now support training with categorical data.  Also, partition-based
```

> categorical split is featured in this release. This split type is first available in

Reviewer: ```suggestion
categorical split is introduced in this release. This split type is first available in
```

> LightGBM in the context of gradient boosting. In the previous version, only `gpu_hist`
> supports one-hot encoding-based split which has the form of `x \in {c}` where `{c}` is the
> set of all categories. In this new release, the `{c}` can be optionally split into 2 sets
> for the left and right nodes using any of the aforementioned tree methods. For more

Reviewer: ```suggestion
LightGBM in the context of gradient boosting. The previous XGBoost release supported one-hot split where the splitting criteria is of form `x \in {c}`, i.e. the categorical feature `x` is tested against a single candidate. The new release allows for more expressive conditions: `x \in S` where the categorical feature `x` is tested against multiple candidates. Moreover, it is now possible to use any tree algorithms (`hist`, `approx`, `gpu_hist`) when creating categorical splits. For more
```

XGBoost 1.6 features initial support for the multi-output model, which includes
multi-output regression and multi-label classification. Along with this, the XGBoost
classifier has proper support for base margin without to need for the user to flatten the
input. In this initial support, XGBoost builds one model for each target similar to the
sklearn meta estimator, for more details, please see our [quick
introduction](https://xgboost.readthedocs.io/en/latest/tutorials/multioutput.html).

(#7365, #7736, #7607, #7574, #7521, #7514, #7456, #7453, #7455, #7434, #7429, #7405, #7381)

### External memory support
External memory support for both approx and hist tree method is considered feature
complete in XGBoost 1.6.  Building upon the iterator-based interface introduced in the
previous version, now both `hist` and `approx` iterates over each batch of data during
training and prediction.  In previous versions, `hist` concatenates all the batches into
an internal representation, which is removed in this version.  As a result, users can
expect higher scalability in terms of data size but might experience lower performance due
to disk IO. (#7531, #7320, #7638, #7372)

### Rewritten approx

> The `approx` tree method is rewritten based on the existing `hist` tree method, the

Reviewer: ```suggestion
The `approx` tree method is rewritten based on the existing `hist` tree method. The
```

rewrite closes the feature gap between `approx` and `hist` and improves the performance.

> Now the behavior and `approx` should be more aligned with `hist` and `gpu_hist`, here's a

Reviewer: ```suggestion
Now the behavior of `approx` should be more aligned with `hist` and `gpu_hist`. Here is a
```
